packages/block-editor/src/store/selectors.js

```diff
@@ -463,6 +464,48 @@ export function getBlockHierarchyRootClientId( state, clientId ) {
 	return parent;
 }
 
+/**
+ * Given a block client ID, returns the next element of the hierarchy from which the block is nested which should be selected onFocus, return the block itself for root level blocks.
```

Comment:
This method should return the common ancestor as the name says, not the first child of the common ancestor.

Reply:
Right!

Reply:
done